iothub/device/src/Transport/ProtocolRoutingDelegatingHandler.cs

```diff
@@ -66,6 +66,7 @@ public override async Task OpenAsync(CancellationToken cancellationToken)
 
                 try
                 {
+                    CreateNewTransportIfNotReady();
```

Comment:
I don't really like the addition of this function as it shows TransportHandler.OpenAsync is not callable again after failure. This should be fixed to all transports in the future.

Reply:
Note that the failure need not be in OpenAsync. This simply says whether the Transport is in a bad state or not. If it is not, then you can call OpenAsync on it. 
We can extend this to do what you are saying by having MqttTransportHandler not get into Error state when OpenAsync fails. This will still work in that case.
So, in my mind, this is a most generic check.